pkg/controller/import-controller_test.go

```diff
@@ -638,6 +638,17 @@ var _ = Describe("Update PVC from POD", func() {
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	It("Should not mark PVC as waiting for VDDK configmap, if image URL annotation is present", func() {
+		pvc := createPvcInStorageClass("testPvc1", "default", &testStorageClass, map[string]string{AnnEndpoint: testEndPoint, AnnImportPod: "testpod", AnnSource: SourceVDDK, AnnVddkInitImageURL: "test://image"}, nil, corev1.ClaimBound)
+		reconciler = createImportReconciler(pvc)
+		err := reconciler.createImporterPod(pvc)
+		Expect(err).ToNot(HaveOccurred())
+		resPvc := &corev1.PersistentVolumeClaim{}
```

Comment:
Should we be looking for the pod, not the pvc here? We created the PVC above, then call the reconcile createImporterPod, but then check the PVC, should be the pod.

Reply:
In this case, createImporterPod is the thing that adds the AwaitingVDDK annotation to the PVC. The test doesn't really care about the pod, it was only created to test that code path through createImporterPod. The real test is to make sure the PVC was updated afterward, as a side effect of attempting to create the pod. So I think the test is basically correct, but maybe there's a better way to do this PVC update.

Now that you point out this test though, I notice that it doesn't actually check for the absence of the AwaitingVDDK condition reason. I will put that in.